landing-page/content/common/release-notes.md

```diff
@@ -68,6 +68,28 @@ To add a dependency on Iceberg in Maven, add the following to your `pom.xml`:
 </dependencies>
 ```
 
+#### Development snapshots
```

Comment:
Can we put this on a different page? We want to keep snapshots and releases separate since we can't recommend testing snapshots to the general public. I think we want it to be documented, though.